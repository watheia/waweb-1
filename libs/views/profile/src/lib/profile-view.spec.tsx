import { render } from '@testing-library/react';

import ProfileView from './profile-view';

describe('waweb.views/profile', () => {
  it('should render successfully', () => {
    const { container } = render(<ProfileView />);
    expect(container).toBeInstanceOf(HTMLDivElement);
    expect(container).toMatchSnapshot();
  });
});
