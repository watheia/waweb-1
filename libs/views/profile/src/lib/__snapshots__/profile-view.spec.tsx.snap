// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`waweb.views/profile should render successfully 1`] = `
<div>
  <div
    class="al flex justify-center px-4 pb-6 sm:px-6 lg:px-8 lg:pb-16"
  >
    <div
      class="max-w-xl overflow-hidden rounded-lg bg-white shadow"
    >
      <form
        action="#"
        class="divide-y divide-gray-200 lg:col-span-12"
        method="POST"
      >
        <div
          class="px-4 py-6 sm:p-6 lg:pb-8"
        >
          <div>
            <h2
              class="text-lg font-medium leading-6 text-gray-900"
            >
              Profile
            </h2>
            <p
              class="mt-1 text-sm text-gray-500"
            >
              This information will be displayed publicly so be careful what you share.
            </p>
          </div>
          <div
            class="mt-6 flex flex-col lg:flex-row"
          >
            <div
              class="flex-grow space-y-6"
            >
              <div>
                <label
                  class="block text-sm font-medium text-gray-700"
                  for="username"
                >
                  Username
                </label>
                <div
                  class="mt-1 flex rounded-md shadow-sm"
                >
                  <span
                    class="inline-flex items-center rounded-l-md border border-r-0 border-gray-300 bg-gray-50 px-3 text-gray-500 sm:text-sm"
                  >
                    workcation.com/
                  </span>
                  <input
                    autocomplete="username"
                    class="block w-full min-w-0 flex-grow rounded-none rounded-r-md border-gray-300 focus:border-sky-500 focus:ring-sky-500 sm:text-sm"
                    id="username"
                    name="username"
                    type="text"
                    value="deblewis"
                  />
                </div>
              </div>
              <div>
                <label
                  class="block text-sm font-medium text-gray-700"
                  for="about"
                >
                  About
                </label>
                <div
                  class="mt-1"
                >
                  <textarea
                    class="mt-1 block w-full rounded-md border border-gray-300 shadow-sm focus:border-sky-500 focus:ring-sky-500 sm:text-sm"
                    id="about"
                    name="about"
                    rows="3"
                  />
                </div>
                <p
                  class="mt-2 text-sm text-gray-500"
                >
                  Brief description for your profile. URLs are hyperlinked.
                </p>
              </div>
            </div>
            <div
              class="mt-6 flex-grow lg:mt-0 lg:ml-6 lg:flex-shrink-0 lg:flex-grow-0"
            >
              <p
                aria-hidden="true"
                class="text-sm font-medium text-gray-700"
              >
                Photo
              </p>
              <div
                class="mt-1 lg:hidden"
              >
                <div
                  class="flex items-center"
                >
                  <div
                    aria-hidden="true"
                    class="inline-block h-12 w-12 flex-shrink-0 overflow-hidden rounded-full"
                  >
                    <img
                      alt=""
                      class="h-full w-full rounded-full"
                      src="https://kgkcfphpovhbptsscyyj.supabase.co/storage/v1/object/public/avatars/supabot.png"
                    />
                  </div>
                  <div
                    class="ml-5 rounded-md shadow-sm"
                  >
                    <div
                      class="group relative flex items-center justify-center rounded-md border border-gray-300 px-3 py-2 focus-within:ring-2 focus-within:ring-sky-500 focus-within:ring-offset-2 hover:bg-gray-50"
                    >
                      <label
                        class="pointer-events-none relative text-sm font-medium leading-4 text-gray-700"
                        for="mobile-user-photo"
                      >
                        <span>
                          Change
                        </span>
                        <span
                          class="sr-only"
                        >
                           user photo
                        </span>
                      </label>
                      <input
                        class="absolute h-full w-full cursor-pointer rounded-md border-gray-300 opacity-0"
                        id="mobile-user-photo"
                        name="user-photo"
                        type="file"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="relative hidden overflow-hidden rounded-full lg:block"
              >
                <img
                  alt=""
                  class="relative h-40 w-40 rounded-full"
                  src="https://kgkcfphpovhbptsscyyj.supabase.co/storage/v1/object/public/avatars/supabot.png"
                />
                <label
                  class="absolute inset-0 flex h-full w-full items-center justify-center bg-black bg-opacity-75 text-sm font-medium text-white opacity-0 focus-within:opacity-100 hover:opacity-100"
                  for="desktop-user-photo"
                >
                  <span>
                    Change
                  </span>
                  <span
                    class="sr-only"
                  >
                     user photo
                  </span>
                  <input
                    class="absolute inset-0 h-full w-full cursor-pointer rounded-md border-gray-300 opacity-0"
                    id="desktop-user-photo"
                    name="user-photo"
                    type="file"
                  />
                </label>
              </div>
            </div>
          </div>
          <div
            class="mt-6 grid grid-cols-12 gap-6"
          >
            <div
              class="col-span-12 sm:col-span-6"
            >
              <label
                class="block text-sm font-medium text-gray-700"
                for="first-name"
              >
                First name
              </label>
              <input
                autocomplete="given-name"
                class="mt-1 block w-full rounded-md border border-gray-300 px-3 py-2 shadow-sm focus:border-sky-500 focus:outline-none focus:ring-sky-500 sm:text-sm"
                id="first-name"
                name="first-name"
                type="text"
              />
            </div>
            <div
              class="col-span-12 sm:col-span-6"
            >
              <label
                class="block text-sm font-medium text-gray-700"
                for="last-name"
              >
                Last name
              </label>
              <input
                autocomplete="family-name"
                class="mt-1 block w-full rounded-md border border-gray-300 px-3 py-2 shadow-sm focus:border-sky-500 focus:outline-none focus:ring-sky-500 sm:text-sm"
                id="last-name"
                name="last-name"
                type="text"
              />
            </div>
            <div
              class="col-span-12"
            >
              <label
                class="block text-sm font-medium text-gray-700"
                for="url"
              >
                URL
              </label>
              <input
                class="mt-1 block w-full rounded-md border border-gray-300 px-3 py-2 shadow-sm focus:border-sky-500 focus:outline-none focus:ring-sky-500 sm:text-sm"
                id="url"
                name="url"
                type="text"
              />
            </div>
            <div
              class="col-span-12 sm:col-span-6"
            >
              <label
                class="block text-sm font-medium text-gray-700"
                for="company"
              >
                Company
              </label>
              <input
                autocomplete="organization"
                class="mt-1 block w-full rounded-md border border-gray-300 px-3 py-2 shadow-sm focus:border-sky-500 focus:outline-none focus:ring-sky-500 sm:text-sm"
                id="company"
                name="company"
                type="text"
              />
            </div>
          </div>
        </div>
        <div
          class="divide-y divide-gray-200 pt-6"
        >
          <div
            class="px-4 sm:px-6"
          >
            <div>
              <h2
                class="text-lg font-medium leading-6 text-gray-900"
              >
                Privacy
              </h2>
              <p
                class="mt-1 text-sm text-gray-500"
              >
                Ornare eu a volutpat eget vulputate. Fringilla commodo amet.
              </p>
            </div>
            <ul
              class="mt-2 divide-y divide-gray-200"
              role="list"
            >
              <li
                class="flex items-center justify-between py-4"
              >
                <div
                  class="flex flex-col"
                >
                  <p
                    class="text-sm font-medium text-gray-900"
                    id="headlessui-label-1"
                  >
                    Available to hire
                  </p>
                  <p
                    class="text-sm text-gray-500"
                    id="headlessui-description-2"
                  >
                    Nulla amet tempus sit accumsan. Aliquet turpis sed sit lacinia.
                  </p>
                </div>
                <button
                  aria-checked="true"
                  aria-describedby="headlessui-description-2"
                  aria-labelledby="headlessui-label-1"
                  class="bg-teal-500 relative ml-4 inline-flex h-6 w-11 flex-shrink-0 cursor-pointer rounded-full border-2 border-transparent transition-colors duration-200 ease-in-out focus:outline-none focus:ring-2 focus:ring-sky-500 focus:ring-offset-2"
                  id="headlessui-switch-3"
                  role="switch"
                  tabindex="0"
                  type="button"
                >
                  <span
                    aria-hidden="true"
                    class="translate-x-5 inline-block h-5 w-5 transform rounded-full bg-white shadow ring-0 transition duration-200 ease-in-out"
                  />
                </button>
              </li>
              <li
                class="flex items-center justify-between py-4"
              >
                <div
                  class="flex flex-col"
                >
                  <p
                    class="text-sm font-medium text-gray-900"
                    id="headlessui-label-4"
                  >
                    Make account private
                  </p>
                  <p
                    class="text-sm text-gray-500"
                    id="headlessui-description-5"
                  >
                    Pharetra morbi dui mi mattis tellus sollicitudin cursus pharetra.
                  </p>
                </div>
                <button
                  aria-checked="false"
                  aria-describedby="headlessui-description-5"
                  aria-labelledby="headlessui-label-4"
                  class="bg-gray-200 relative ml-4 inline-flex h-6 w-11 flex-shrink-0 cursor-pointer rounded-full border-2 border-transparent transition-colors duration-200 ease-in-out focus:outline-none focus:ring-2 focus:ring-sky-500 focus:ring-offset-2"
                  id="headlessui-switch-6"
                  role="switch"
                  tabindex="0"
                  type="button"
                >
                  <span
                    aria-hidden="true"
                    class="translate-x-0 inline-block h-5 w-5 transform rounded-full bg-white shadow ring-0 transition duration-200 ease-in-out"
                  />
                </button>
              </li>
              <li
                class="flex items-center justify-between py-4"
              >
                <div
                  class="flex flex-col"
                >
                  <p
                    class="text-sm font-medium text-gray-900"
                    id="headlessui-label-7"
                  >
                    Allow commenting
                  </p>
                  <p
                    class="text-sm text-gray-500"
                    id="headlessui-description-8"
                  >
                    Integer amet, nunc hendrerit adipiscing nam. Elementum ame
                  </p>
                </div>
                <button
                  aria-checked="true"
                  aria-describedby="headlessui-description-8"
                  aria-labelledby="headlessui-label-7"
                  class="bg-teal-500 relative ml-4 inline-flex h-6 w-11 flex-shrink-0 cursor-pointer rounded-full border-2 border-transparent transition-colors duration-200 ease-in-out focus:outline-none focus:ring-2 focus:ring-sky-500 focus:ring-offset-2"
                  id="headlessui-switch-9"
                  role="switch"
                  tabindex="0"
                  type="button"
                >
                  <span
                    aria-hidden="true"
                    class="translate-x-5 inline-block h-5 w-5 transform rounded-full bg-white shadow ring-0 transition duration-200 ease-in-out"
                  />
                </button>
              </li>
              <li
                class="flex items-center justify-between py-4"
              >
                <div
                  class="flex flex-col"
                >
                  <p
                    class="text-sm font-medium text-gray-900"
                    id="headlessui-label-10"
                  >
                    Allow mentions
                  </p>
                  <p
                    class="text-sm text-gray-500"
                    id="headlessui-description-11"
                  >
                    Adipiscing est venenatis enim molestie commodo eu gravid
                  </p>
                </div>
                <button
                  aria-checked="true"
                  aria-describedby="headlessui-description-11"
                  aria-labelledby="headlessui-label-10"
                  class="bg-teal-500 relative ml-4 inline-flex h-6 w-11 flex-shrink-0 cursor-pointer rounded-full border-2 border-transparent transition-colors duration-200 ease-in-out focus:outline-none focus:ring-2 focus:ring-sky-500 focus:ring-offset-2"
                  id="headlessui-switch-12"
                  role="switch"
                  tabindex="0"
                  type="button"
                >
                  <span
                    aria-hidden="true"
                    class="translate-x-5 inline-block h-5 w-5 transform rounded-full bg-white shadow ring-0 transition duration-200 ease-in-out"
                  />
                </button>
              </li>
            </ul>
          </div>
          <div
            class="mt-4 flex justify-end px-4 py-4 sm:px-6"
          >
            <button
              class="inline-flex justify-center rounded-md border border-gray-300 bg-white px-4 py-2 text-sm font-medium text-gray-700 shadow-sm hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-sky-500 focus:ring-offset-2"
              type="button"
            >
              Cancel
            </button>
            <button
              class="ml-5 inline-flex justify-center rounded-md border border-transparent bg-sky-700 px-4 py-2 text-sm font-medium text-white shadow-sm hover:bg-sky-800 focus:outline-none focus:ring-2 focus:ring-sky-500 focus:ring-offset-2"
              type="submit"
            >
              Save
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
